### PR TITLE
`azurerm_data_factory_integration_runtime_self_hosted` - fix failed test case

### PR DESCRIPTION
**Pull request by ms-henglu**
To include the fix: https://github.com/Azure/azure-quickstart-templates/pull/12821/files

image

### CODE REVIEW

File: internal/services/datafactory/data_factory_integration_runtime_self_hosted_resource_test.go [hunk 1]
```diff
@@ -73,7 +73,11 @@ resource "azurerm_data_factory_integration_runtime_self_hosted" "test" {
 func (IntegrationRuntimeSelfHostedResource) rbac(data acceptance.TestData) string {
 	return fmt.Sprintf(`
 provider "azurerm" {
-  features {}
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
```
Comment (jackofallops) [inline internal/services/datafactory/data_factory_integration_runtime_self_hosted_resource_test.go:78]: Needing this suggests that a resource is not being correctly deleted. Can we look into what's not being removed and see if there's a delete bug?

**Review by jackofallops — CHANGES_REQUESTED** [review]
HI @ms-henglu 
I'm concerned that the fix here is just masking an underlying problem, can we check what's not being deleted and see if there's a resource that possibly has a delete issue?

Thanks!

**Comment by ms-henglu** [comment]
Hi @jackofallops ,

Thanks for the code review, I've updated the test case, would you please take another look, thanks!

image

**Review by katbyte — APPROVED** [review]
LGTM 🗄️

**Pull request merged**